src/openms_gui/source/VISUAL/TVDIATreeTabController.cpp

```diff
@@ -0,0 +1,214 @@
+//    notice, this list of conditions and the following disclaimer in the
+//    documentation and/or other materials provided with the distribution.
+//  * Neither the name of any author or any participating institution
+//    may be used to endorse or promote products derived from this software
+//    without specific prior written permission.
+// For a full list of authors, refer to the file AUTHORS.
+// --------------------------------------------------------------------------
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL ANY OF THE AUTHORS OR THE CONTRIBUTING
+// INSTITUTIONS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS;
+// OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+// WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR
+// OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF
+// ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// --------------------------------------------------------------------------
+// $Maintainer: Chris Bielow $
+// $Authors: Chris Bielow $
+// --------------------------------------------------------------------------
+
+#include <OpenMS/VISUAL/TVDIATreeTabController.h>
+
+#include <OpenMS/CONCEPT/RAIICleanup.h>
+#include <OpenMS/KERNEL/ChromatogramTools.h>
+#include <OpenMS/KERNEL/OnDiscMSExperiment.h>
+#include <OpenMS/VISUAL/APPLICATIONS/TOPPViewBase.h>
+#include <OpenMS/VISUAL/AxisWidget.h>
+#include <OpenMS/VISUAL/Plot1DWidget.h>
+
+#include <QtWidgets/QMessageBox>
+#include <QtCore/QString>
+
+using namespace OpenMS;
+using namespace std;
+
+namespace OpenMS
+{
+  TVDIATreeTabController::TVDIATreeTabController(TOPPViewBase* parent):
+    TVControllerBase(parent)
+  {  
+  }
+
+  LayerData::ExperimentSharedPtrType prepareChromatogram(Size index, LayerData::ExperimentSharedPtrType exp_sptr, LayerData::ODExperimentSharedPtrType ondisc_sptr)
+  {
+    // create a managed pointer fill it with a spectrum containing the chromatographic data
+    LayerData::ExperimentSharedPtrType chrom_exp_sptr(new LayerData::ExperimentType());
+    chrom_exp_sptr->setMetaValue("is_chromatogram", "true"); //this is a hack to store that we have chromatogram data
+    LayerData::ExperimentType::SpectrumType spectrum;
+
+    // retrieve chromatogram (either from in-memory or on-disc representation)
+    MSChromatogram current_chrom;
+    current_chrom = exp_sptr->getChromatograms()[index];
+    if (current_chrom.empty() )
+    {
+      current_chrom = ondisc_sptr->getChromatogram(index);
+    }
+
+    // fill "dummy" spectrum with chromatogram data
```

Comment:
this hack is so annoying ;) hope we get rid of it at some point

Reply:
:-)